src/middlewared/middlewared/plugins/usage.py

```diff
@@ -46,34 +49,82 @@ async def start(self):
 
         return True
 
-    async def gather(self):
-        network = await self.middleware.call('interfaces.query')
-
-        hardware = await self.gather_hardware(network)
-        jails = await self.gather_jails()
-        network = await self.gather_network(network)
-        system = await self.gather_system()
-        plugins = await self.gather_plugins()
-        pools = await self.gather_pools()
-        services = await self.gather_services()
-        sharing = await self.gather_sharing()
-        vms = await self.gather_vms()
+    def gather(self):
+        datasets = self.middleware.call_sync('zfs.dataset.query')
+        context = {
+            'network': self.middleware.call_sync('interfaces.query'),
+            'root_datasets': {d['id']: d for d in datasets if '/' not in d['id']},
+            'zvols': filter_list(datasets, [['type', '=', 'VOLUME']]),
+            'datasets': {d['id']: d for d in datasets},
+        }
 
         return json.dumps(
             {
-                **hardware,
-                **jails,
-                **network,
-                **{k: v for l in system['gather_system'] for k, v in l.items()},
-                **plugins,
-                **pools,
-                **services,
-                **sharing,
-                **vms
+                k: v for f in dir(self) if f.startswith('gather_') and callable(getattr(self, f))
+                for k, v in (self.middleware.call_sync(f'usage.{f}', context)).items()
             }, sort_keys=True
         )
 
-    async def gather_hardware(self, network):
+    def gather_backup_data(self, context):
+        backed = {
+            'cloudsync': 0,
+            'rsynctask': 0,
+            'zfs_replication': 0,
+            'total_size': 0,
+        }
+        datasets_data = context['datasets']
+        datasets = deepcopy(datasets_data)
+        for namespace in ('cloudsync', 'rsynctask'):
+            task_datasets = deepcopy(datasets_data)
+            for task in self.middleware.call_sync(
+                f'{namespace}.query', [['enabled', '=', True], ['direction', '=', 'PUSH'], ['locked', '=', False]]
+            ):
+                task_ds = self.middleware.call_sync('zfs.dataset.get_dataset_by_path', task['path'], task_datasets)
+                if task_ds:
+                    backed[namespace] += task_datasets.pop(task_ds)['properties']['used']['parsed']
```

Comment:
How do you know task_datasets has not been popped before for task_ds?

Reply:
`get_dataset_by_path` ensures we found the dataset in `task_datasets`